modules/programs/ghostty.nix

```diff
@@ -0,0 +1,71 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.ghostty;
+  keyValueSettings = {
+    listsAsDuplicateKeys = true;
+    mkKeyValue = lib.generators.mkKeyValueDefault { } " = ";
+  };
+  keyValue = pkgs.formats.keyValue keyValueSettings;
+in {
+  meta.maintainers = [ lib.maintainers.HeitorAugustoLN ];
+
+  options.programs.ghostty = {
+    enable = lib.mkEnableOption "ghostty";
+    package = lib.mkPackageOption pkgs "ghostty" { };
+    settings = lib.mkOption {
+      inherit (keyValue) type;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          theme = "catppuccin-mocha";
+          font-size = 10;
+        }
+      '';
+      description = ''
+        Configuration written to $XDG_CONFIG_HOME/ghostty/config.
+
+        See https://ghostty.org/docs/config/reference for more information.
+      '';
+    };
+    clearDefaultKeybinds = lib.mkEnableOption "clear default keybinds" // {
+      description = "Whether to clear default keybinds.";
+    };
+    installBatSyntax =
+      lib.mkEnableOption "installation of ghostty configuration syntax for bat"
+      // {
+        default = true;
+      };
+    installVimSyntax = lib.mkEnableOption
+      "installation of ghostty configuration syntax for vim/neovim";
+  };
+
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      xdg.configFile."ghostty/config" = lib.mkIf (cfg.settings != { }) {
```

Comment:
This is where Linux users would place their config, but for macos users, that location is `home.file."Library/Application Support/com.mitchellh.ghostty/config"`. 

I know Ghostty for darwin-aarch64 isn't part of nixpkgs, but it would still be very convenient for darwin users to be able to configure things though home. WDYT?

Reply:
Thought it were the same, thanks for the information

Reply:
as far as i know macos also support xdg config, right? ghostty docs says the same and even look on both options

Reply:
Mac will use either, and will check XDG_CONFIG_HOME first: https://ghostty.org/docs/config#file-location

Reply:
Thanks folks. I missed that part where XDG_CONFIG_HOME was supported. My bad and sorry for the noise.